### PR TITLE
#606 Implemented and tested Wallet#pay().

### PR DESCRIPTION
**Pull request by criske**
PR for #606

### CODE REVIEW

File: self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                    new MkAnswer.Simple(
+                        HttpURLConnection.HTTP_OK,
+                        Json.createObjectBuilder()
+                            .add("id", "pi_1DiKBG2eZvKYlo2CRcpuIHvu")
+                            .add("created", 1545045758)
+                            .add("status", "succeeded")
+                            .build()
+                            .toString()
+                    )
+            ).start(this.resource.port())
```
Comment (amihaiemil) [inline self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java:76]: @criske Why are you starting a Mock server here? The Stripe API uses static methods, I don't see where you are configuring the URL.

File: self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            Mockito.when(storage.wallets()).thenReturn(allWallets);
+            Mockito.when(allWallets.ofProject(project)).thenReturn(ofProject);
+
+            stripe.pay(invoice);
+
+            Mockito.verify(invoices, Mockito.times(1))
+                .registerAsPaid(Mockito.any(Invoice.class));
+            Mockito.verify(ofProject, Mockito.times(1))
+                .updateCash(stripe, BigDecimal.valueOf(990));
+        }
```
Comment (amihaiemil) [inline self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java:143]: @criske I don't think there is any way to write an efficient test for interaction with Stripe... the SDK always calls the prod URL

File: self-core-impl/src/main/java/com/selfxdsd/core/projects/StripeWallet.java [hunk 3]
```diff
@@ -96,8 +132,97 @@ public BigDecimal cash() {
...
+                );
+            }
+            final PaymentIntent paymentIntent = PaymentIntent
+                .create(PaymentIntentCreateParams.builder()
+                    .setCurrency("usd")
+                    .setAmount(invoice.totalAmount().longValueExact())
+                    .setCustomer(payoutMethod.identifier())
+                    .setPaymentMethod(paymentMethod.identifier())
+                    .setConfirm(true)
+                    .build());
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/projects/StripeWallet.java:183]: @criske Which Stripe Docs did you read in order to come up with this code? :D

**Review by amihaiemil — COMMENTED** [review]
@criske Looks ok, although the PR is too big... :D small PRs please, don't forget.

File: self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java (hunk 1 above)
Comment (criske) [reply self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java:76]: @amihaiemil Mock server url is configured at line 59 `Stripe.overrideApiBase(container.home().toString());`

File: self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java (hunk 2 above)
Comment (criske) [reply self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java:143]: @amihaiemil Just needed a "successful" status to verify the Invoice payment logic.

File: self-core-impl/src/main/java/com/selfxdsd/core/projects/StripeWallet.java (hunk 3 above)
Comment (criske) [reply self-core-impl/src/main/java/com/selfxdsd/core/projects/StripeWallet.java:183]: @amihaiemil [Payment Intents API](https://stripe.com/docs/payments/payment-intents)
https://stripe.com/docs/api/payment_intents/create
https://stripe.com/docs/api/payment_intents/object

File: self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java (hunk 2 above)
Comment (amihaiemil) [reply self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java:143]: @criske right, my bad. I missed the "overrideApiBase" method

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**